geode-management/src/test/java/org/apache/geode/management/api/ClusterManagementOperationResultTest.java

```diff
@@ -37,20 +39,32 @@ public void setUp() {
 
   @Test
   public void serialize() throws Exception {
-    CompletableFuture<TestOperationResult> operationResult =
-        new CompletableFuture<>();
     ClusterManagementResult result1 = new ClusterManagementResult();
     result1.setStatus(StatusCode.OK, "Success!!");
-    ClusterManagementOperationResult<TestOperationResult> result =
-        new ClusterManagementOperationResult<>(result1, operationResult, new Date(),
-            new CompletableFuture<>(), "operator", "id");
+    ClusterManagementOperationResult<ClusterManagementOperation<OperationResult>, OperationResult> result =
+        new ClusterManagementOperationResult<>(result1, new Date(), new Date(),
+            null, "id", null, null);
     String json = mapper.writeValueAsString(result);
     System.out.println(json);
-    ClusterManagementOperationResult value =
+    ClusterManagementOperationResult<ClusterManagementOperation<OperationResult>, OperationResult> value =
         mapper.readValue(json, ClusterManagementOperationResult.class);
     assertThat(value.getStatusMessage()).isEqualTo("Success!!");
   }
 
-  static class TestOperationResult implements OperationResult {
+  @Test
+  public void serializeRebal() throws Exception {
+    ClusterManagementResult result1 = new ClusterManagementResult();
+    result1.setStatus(StatusCode.OK, "Success!!");
+    RebalanceOperation rebalanceOperation = new RebalanceOperation();
+    rebalanceOperation.setOperator("operator");
+    ClusterManagementOperationResult<RebalanceOperation, RebalanceResult> result =
+        new ClusterManagementOperationResult(result1, new Date(), new Date(),
+            rebalanceOperation, "id", new RebalanceResultImpl(), null);
```

Comment:
Yes, let us look at that. Agree that given that we're extending an interface already in `RebalanceResult` it makes sense to make `RebalanceResult` the implemenation.

Reply:
This PR did not introduce RebalanceResult nor RebalanceResultImpl.
I prefer keeping the current interface and having an internal implementation class. This prevents our external apis from having internal things exposed. For example, in this case, the external api does not need a constructor nor settors.
I think it would be good to have more of our external api contract defined by interfaces and try to minimize the number of classes in it.

Reply:
ps: I'm fine with some other name for the class that implements the RebalanceResult.
I also find it odd that RebalanceResult has "getSuccess()" and "getStatusMessage()" but its super interface "OperationResult" is empty. It seems like both of these would be reasonable things to expect from every OperationResult.

Reply:
I filed a geode ticket about moving those methods up to OperationResult.
The team talked and we decided it would be best to keep the impl class